compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -292,6 +298,86 @@ impl<'hir> LoweringContext<'_, 'hir> {
         }
     }
 
+    /// Checks if an expression refers to a function marked with
+    /// `#[rustc_legacy_const_generics]` and returns the argument index list
+    /// from the attribute.
+    fn legacy_const_generic_args(&mut self, expr: &Expr) -> Option<Vec<usize>> {
+        if let ExprKind::Path(None, path) = &expr.kind {
+            if path.segments.last().unwrap().args.is_some() {
```

Comment:
Please leave a comment stating that this rewrite doesn't work for function calls with generic parameters specified.

Isn't this a problem as some of these functions already have type generics and there may be stable code out there in the wild that has calls with generic types specified?

Reply:
None of the functions using `rustc_args_required_const` are generic, they are all based on a vendor's C API.

Excluding generics is intentional: if you're trying to use generics then you are most likely trying to call the "new" form of the intrinsic and the rewrite shouldn't apply. I'll add a comment to clarify.